parlai/core/torch_agent.py

```diff
@@ -96,22 +88,20 @@ class Batch(AttrDict):
 
     :param image:
         list of image features in the format specified by the --image-mode arg.
-
-    :param observations:
-        the original observations in the batched order
     """
 
     batchsize: int
     text_vec: Optional[torch.LongTensor]
-    text_lengths: Optional[List[int]]
     label_vec: Optional[torch.LongTensor]
-    label_lengths: Optional[List[int]]
     labels: Optional[List[str]]
-    valid_indices: Optional[List[int]]
+    valid_indices: Optional[torch.LongTensor]
     candidates: Optional[List[List[str]]]
     candidate_vecs: Optional[List[List[torch.LongTensor]]]
     image: Optional[List[Any]]
-    observations: Optional[List[Message]]
+    _context_original_length: Optional[torch.LongTensor]
+    _context_truncate_rate: Optional[torch.LongTensor]
+    _label_original_length: Optional[torch.LongTensor]
+    _label_truncate_rate: Optional[torch.LongTensor]
 
     def __init__(
```

Comment:
perhaps we can warn here if we're passing in a non-tensor?

Reply:
For specifically these 4 or others? We have non-tensors (bools, Nones, etc).

Reply:
wait so... we're not fully getting rid of non-tensors here?

Reply:
Nope. I benchmarked them and they're not painful. Only the complex objects are.

Reply:
how complex is complex? does this mean I can just make my own batch object and keep strings in there?

Reply:
Ya I mean, we don't have any hard limitation. You can even do full observations if you want, you'll just pay a penalty with background workers. But any Batch object you _manually_ want to add things to is still allowed.

I'll leave notes in batch descriptions